models/issue_comment.go

```diff
@@ -673,6 +808,58 @@ func CreateIssueComment(doer *User, repo *Repository, issue *Issue, content stri
 	return comment, nil
 }
 
+// CreateCodeComment creates a plain code comment at the specified line / path
+func CreateCodeComment(doer *User, repo *Repository, issue *Issue, content, treePath string, line, reviewID int64) (*Comment, error) {
+	var commitID, patch string
+	pr, err := GetPullRequestByIssueID(issue.ID)
+	if err != nil {
+		return nil, fmt.Errorf("GetPullRequestByIssueID: %v", err)
+	}
+	if err := pr.GetHeadRepo(); err != nil {
+		return nil, fmt.Errorf("GetHeadRepo: %v", err)
+	}
+	gitRepo, err := git.OpenRepository(pr.HeadRepo.RepoPath())
+	if err != nil {
+		return nil, fmt.Errorf("OpenRepository: %v", err)
+	}
+	// FIXME differentiate between previous and proposed line
+	var gitLine = line
+	if gitLine < 0 {
+		gitLine *= -1
+	}
+	// FIXME validate treePath
+	// Get latest commit referencing the commented line
+	commit, err := gitRepo.LineBlame(pr.HeadBranch, gitRepo.Path, treePath, uint(gitLine))
+	if err != nil {
+		return nil, fmt.Errorf("LineBlame[%s, %s, %s, %d]: %v", pr.HeadBranch, gitRepo.Path, treePath, gitLine, err)
+	}
+	// Only fetch diff if comment is review comment
+	if reviewID != 0 {
+		headCommitID, err := gitRepo.GetRefCommitID(pr.GetGitRefName())
```

Comment:
```
[.../repo/pull_review.go:79 CreateCodeComment()] [E] CreateCodeComment: GetRefCommitID[refs/pull/2/head]: object does not exist [id: refs/pull/2/head, rel_path: ]
```
Error come from this line. We should use `baseGitRepo` but not `headGitRepo` here I think.

Reply:
I've changed the discussed code. Could you please make sure that it is fixed now? Your bug only occurs if the PR comes from another repo (e.g. fork).